source/simulator/core.cc

```diff
@@ -1043,6 +1046,7 @@ namespace aspect
             {
               coupling[x.velocities[d]][x.pressure] = DoFTools::always;
               coupling[x.pressure][x.velocities[d]] = DoFTools::always;
+              coupling[x.pressure][x.pressure] = DoFTools::always;
```

Comment:
You discover one of the reasons why I think that that benchmark is so poorly posed :-( It just doesn't fit into the typical finite element method unless you have a specially constructed code and a particular mesh that resolves the interface. I don't think the problem is mathematically well posed.

In this benchmark, the velocity needs to be continuous -- you prescribe it in parts of the domain and that forms velocity boundary conditions for the rest of the domain. But in general, the pressure will be discontinuous for this setup: if there is no gravity, then in the "special" part of the domain where you have rigid body motion, the pressure will be constant, whereas in general it will of course no be constant along the boundary of the special domain and may in fact be singular (in corners). 

I think this benchmark is poorly designed and mathematically not well posed. I've refused to touch it for the past decade since learning about it from Scott, but I recognize that others may nevertheless want to reproduce it. (I'm going to go on the record, though, saying that you can't expect good convergence from ASPECT unless you align the mesh with the interface.) The question is just how we can make this work in ASPECT without distorting ourselves and putting too many things into the code that are only applicable to this kind of benchmark. If there is a signal we can attach code to in a plugin that's specific to this benchmark, that's ok with me. But I'd rather not us change a core piece of the code just to accommodate this one benchmark because that's a slope I don't want us to get on.